projects/fmcjesdadc1/src/app/fmcjesdadc1.c

```diff
@@ -67,20 +68,33 @@ int main(void)
 {
 
 	int32_t status;
-
 	// SPI configuration
-	struct spi_init_param ad9250_spi_param = {
+	struct spi_init_param demux_spi_param = {
 		.max_speed_hz = 2000000u,
 		.chip_select = 0,
 		.mode = SPI_MODE_0,
 		.platform_ops = &xil_platform_ops
 	};
 
-	struct spi_init_param ad9517_spi_param = {
+	struct spi_init_param ad9250_0_spi_param = {
 		.max_speed_hz = 2000000u,
```

Comment:
I think the `.max_speed_hz` and `.mode` are only relevant in the `demux_spi_param` and not for the devices themselves. You could remove them from here to avoid confusion.

Reply:
One other idea would be to force the user, with some checks in demux code, to put identical values for `.mode` and `.max_speed_hz` because the demux that we implement here only supports slave devices with different chip selects but with identical mode and speed.

Reply:
> 
> 
> I think the `.max_speed_hz` and `.mode` are only relevant in the `demux_spi_param` and not for the devices driven by demux themselves. You could remove the initialization from here to avoid confusion.

I agree that these parameters are irrelevant for the devices, but I would not let them uninitialized.
For a better understanding and avoiding confusion, I would initialize them with the parameters provided by `demux_spi_init`.

Something like this:
`struct spi_init_param ad9250_0_spi_param = { .max_speed_hz = demux_spi_param.max_speed_hz, .mode = demux_spi_param.mode, `

Reply:
Makes sense, it shows that device clocks and mode cannot be anything else than demux clock /mode.